docs/ui/components/Text/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { css, SerializedStyles } from '@emotion/react';
+import { theme, typography } from '@expo/styleguide';
+import React, { HTMLAttributes } from 'react';
+
+import { Link, LinkProps } from '~/ui/components/Link';
+import { durations } from '~/ui/foundations/durations';
+
+export enum TextElement {
+  CODE = 'code',
+  H1 = 'h1',
+  H2 = 'h2',
+  H3 = 'h3',
+  H4 = 'h4',
+  H5 = 'h5',
+  H6 = 'h6',
+  LI = 'li',
+  P = 'p',
+  SPAN = 'span',
+  UL = 'ul',
+  OL = 'ol',
+  PRE = 'pre',
+}
+
+type TextComponentProps = HTMLAttributes<
+  HTMLHeadingElement | HTMLParagraphElement | HTMLLIElement | HTMLUListElement | HTMLPreElement
+> & {
+  testID?: string;
+  size?: 'small';
+  weight?: 'regular' | 'medium' | 'semiBold' | 'black';
+  theme?: 'secondary' | 'success' | 'warning' | 'error';
+  tag?:
+    | 'code'
+    | 'h1'
+    | 'h2'
+    | 'h3'
+    | 'h4'
+    | 'h5'
+    | 'h6'
+    | 'li'
+    | 'p'
+    | 'span'
+    | 'ul'
+    | 'ol'
+    | 'pre';
+};
+
+export function createTextComponent(Element: TextElement, textStyle?: SerializedStyles) {
+  function TextComponent(props: TextComponentProps) {
+    const { className, testID, size, weight, tag, theme, ...rest } = props;
+    const TextElementTag = tag ?? Element;
+
+    function getSizeStyle() {
+      if ((Element === TextElement.P || Element === TextElement.SPAN) && size === 'small') {
+        return css(typography.body.callout);
+      }
+      return null;
+    }
```

Comment:
If we dont have any instances of `<P size="small">` anywhere, we should remove this. (This was a migration holdover in the website so that I could put off changing a few hundred files when implementing the new typography styles.)

Reply:
Cool, I was a bit confused by this too. Didn't want to randomly strip things out that might be required later on. Will remove this!